prog/hints_test.go

```diff
@@ -44,21 +46,70 @@ func TestHintsCheckConstArg(t *testing.T) {
 		{
 			"Multiple replacers test",
 			0xabcd,
+			2,
 			CompMap{0xabcd: uint64Set{0x2: true, 0x3: true}},
 			[]uint64{0x2, 0x3},
 		},
 		// Checks that special ints are not used.
 		{
 			"Special ints test",
 			0xabcd,
+			2,
 			CompMap{0xabcd: uint64Set{0x1: true, 0x2: true}},
 			[]uint64{0x2},
 		},
 	}
+	runHintsConstArgTests(tests, t)
+}
+
+// Checks if new value fits into field during hints mutation. If the replacer
+// has a higher value, it will be truncated.
+func TestHintsCheckSizeConstArg(t *testing.T) {
+	t.Parallel()
+	var tests = []ConstArgTest{
+		{
+			"Int8 argument test - invalid assigned value",
+			0x12ab,
+			1,
+			CompMap{
+				0xab:   uint64Set{0xab: true, 0xac: true, 0xabcd: true},
+				0x12ab: uint64Set{0xa1: true, 0xaa2: true, 0xaaa3: true},
+				0xffab: uint64Set{0x11: true, 0x22: true, 0x1234: true},
```

Comment:
I think we can potentially got 0xffab from 0xab if it was sign-extended to short.